client/ayon_core/pipeline/farm/pyblish_functions.py

```diff
@@ -495,34 +491,41 @@ def _get_real_frames_to_render(frames):
     return frames_to_render
 
 
-def _get_real_files_to_rendered(collection, frames_to_render):
-    """Use expected files based on real frames_to_render.
+def _get_real_files_to_render(collection, frames_to_render):
+    """Filter files with frames that should be really rendered.
 
-    Artists might explicitly set frames they want to render via Publisher UI.
-    This uses this value to filter out files
-    Args:
-        frames_to_render (list): of str '1001'
-    """
-    files = [os.path.basename(f) for f in list(collection)]
-    file_name, extracted_frame = list(collect_frames(files).items())[0]
+    'expected_files' are collected from DCC based on timeline setting. This is
+    being calculated differently in each DCC. Filtering here is on single place
 
-    if not extracted_frame:
-        return files
+    But artists might explicitly set frames they want to render in Publisher UI
+    This range would override and filter previously prepared expected files
+    from DCC.
 
-    found_frame_pattern_length = len(extracted_frame)
-    normalized_frames_to_render = {
-        str(frame_to_render).zfill(found_frame_pattern_length)
-        for frame_to_render in frames_to_render
-    }
+    Args:
+        collection (clique.Collection): absolute paths
+        frames_to_render (list[int]): of int 1001
+    Returns:
+        (list[str])
 
-    return [
-        file_name
-        for file_name in files
-        if any(
-            frame in file_name
-            for frame in normalized_frames_to_render
-        )
+    Example:
+    --------
+
+    expectedFiles = [
+        "foo_v01.0001.exr",
+        "foo_v01.0002.exr",
     ]
+    frames_to_render = '0001'
```

Comment:
I think this example value is wrong because according to the type hints it's not supposed to be a `str` but `list[int]`?